components/scream/src/diagnostics/tests/longwave_cloud_forcing_tests.cpp

```diff
@@ -0,0 +1,178 @@
+
+  // A time stamp
+  util::TimeStamp t0 ({2022,1,1},{0,0,0});
+
+  // Construct the Diagnostic
+  ekat::ParameterList params;
+  params.set<std::string>("Diagnostic Name", "Longwave Cloud Forcing");
+  params.set<std::string>("Grid", "Point Grid");
+  register_diagnostics();
+  auto& diag_factory = AtmosphereDiagnosticFactory::instance();
+  auto diag = diag_factory.create("LongwaveCloudForcing",comm,params);
+  diag->set_grids(gm);
+
+
+  // Set the required fields for the diagnostic.
+  std::map<std::string,Field> input_fields;
+  for (const auto& req : diag->get_required_field_requests()) {
+    Field f(req.fid);
+    auto & f_ap = f.get_header().get_alloc_properties();
+    f_ap.request_allocation(packsize);
+    f.allocate_view();
+    const auto name = f.name();
+    f.get_header().get_tracking().update_time_stamp(t0);
+    diag->set_required_field(f.get_const());
+//    REQUIRE_THROWS(diag->set_computed_field(f));
+    input_fields.emplace(name,f);
+  }
+
+  // Initialize the diagnostic
+  diag->initialize(t0,RunType::Initial);
+
+  // Run tests
+  {
+    // Construct random data to use for test
+    // Get views of input data and set to random values
+    const auto& LW_flux_up_f          = input_fields["LW_flux_up"];
+    const auto& LW_flux_up_v          = LW_flux_up_f.get_view<Pack**>();
+    const auto& LW_clrsky_flux_up_f          = input_fields["LW_clrsky_flux_up"];
+    const auto& LW_clrsky_flux_up_v          = LW_clrsky_flux_up_f.get_view<Pack**>();
+
+    for (int icol=0;icol<ncols;icol++) {
+      const auto& LW_flux_up_sub      = ekat::subview(LW_flux_up_v,icol);
+      const auto& LW_clrsky_flux_up_sub      = ekat::subview(LW_clrsky_flux_up_v,icol);
+//      const auto& dp_sub      = ekat::subview(pseudo_dens_v,icol);
+//      ekat::genRandArray(dview_as_real(pseudo_density),  engine, pdf_pseudodens);
+//      Kokkos::deep_copy(dp_sub,pseudo_density);
+
+      ekat::genRandArray(dview_as_real(LW_flux_up),              engine, pdf_LW_flux_x);
+      ekat::genRandArray(dview_as_real(LW_clrsky_flux_up),              engine, pdf_LW_flux_x);
+      Kokkos::deep_copy(LW_flux_up_sub,LW_flux_up);
+      Kokkos::deep_copy(LW_clrsky_flux_up_sub,LW_clrsky_flux_up);
+
+    }
+   // ekat::genRandArray(phis_v, engine, pdf_surface);
+
+    // Run diagnostic and compare with manual calculation
+    diag->compute_diagnostic();
+    const auto& diag_out = diag->get_diagnostic();
+    Field LWCF_f = diag_out.clone();
+    LWCF_f.deep_copy<double,Host>(0.0);
+    LWCF_f.sync_to_dev();
+    const auto& LWCF_v = LWCF_f.get_view<Real*>();
+    const int pack_surf = std::min(num_levs / Pack::n, num_mid_packs-1);
+    const int idx_surf  = num_levs % Pack::n;
+    Kokkos::parallel_for("", policy, KOKKOS_LAMBDA(const MemberType& team) {
+      const int icol = team.league_rank();
+      LWCF_v(icol) = LW_clrsky_flux_up_v(icol,0)[0] -  LW_flux_up_v(icol,0)[0] ;
+    });
+    Kokkos::fence();
+    REQUIRE(views_are_equal(diag_out,LWCF_f));
+  }
```

Comment:
Are there any other physical or mathematical based unit tests we can add here?

For example, if we set `LW_clrsky_flux_up_v = LW_flux_up_v` then `LWCF_v=0` for all cols.
If we set `LW_flux_up_v = -LW_clrsky_flux_up_v` then `LWCF_v=2*LW_clrsky_flux_up_v(icol,0)[0]`